### PR TITLE
ER-320 employer dashboard filtering

### PR DESCRIPTION
**Pull request by sachpatel**
Have met the AC. Will need @sdholder to fix some styling alignment issues and markup cleanup 🙏 Lifted the markup from the prototype linked to from the JIRA ticket.

### CODE REVIEW

File: src/Employer/Employer.Web/Controllers/DashboardController.cs [hunk 1]
```diff
@@ -1,31 +1,46 @@
...
             
             if (TempData.ContainsKey(TempDataKeys.DashboardErrorMessage))
             {
                 vm.WarningMessage = TempData[TempDataKeys.DashboardErrorMessage].ToString();
             }
+            
+            if (!string.IsNullOrEmpty(statusFilter) && Enum.TryParse<VacancyStatus>(statusFilter, out var filter))
+            {
+                vm.Filter = filter;
+                Response.Cookies.Append(CookieNames.VacancyStatusFilter, filter.GetDisplayName(), EsfaCookieOptions.GetDefaultHttpCookieOption(_hostingEnvironment));
```
Comment (chrisjensenuk) [inline src/Employer/Employer.Web/Controllers/DashboardController.cs]: Should this be filter.ToString()? GetDisplayName returns human readable string.

File: src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml [hunk 2]
```diff
@@ -1,43 +1,113 @@
...
+                <div class="data lottery-ticket">
+                    <a asp-route="@RouteNames.Dashboard_Index_Get" class="prototype-link">
+                        <h2 class="bold-xlarge">@Model.NoOfVacancies</h2>
+                        <p class="bold-xsmall">All</p>
+                    </a>
+                </div>
+            </div>
+            <div class="column-one-quarter">
+                <div class="data lottery-ticket">
+                    <a asp-route="@RouteNames.Dashboard_Index_Get" asp-route-statusFilter="@VacancyStatus.Draft.GetDisplayName()" class="prototype-link">
```
Comment (chrisjensenuk) [inline src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml]: Draft.ToString()?

File: src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml [hunk 3]
```diff
@@ -1,43 +1,113 @@
...
+                    <th class="bold">Reference number</th>
                     <th class="bold">Status</th>
                 </tr>
             </thead>
             <tbody>
-                @foreach (var vacancy in Model.Vacancies)
-                {
+@foreach (var vacancy in Model.DisplayVacancies)
+{
+    var isMissingVacancyReferenceNo = !vacancy.HasVacancyReference;
```
Comment (chrisjensenuk) [inline src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml]: I don't think this variable adds any value

File: src/Employer/Employer.Web/Views/Part1/Employer/Employer.cshtml [hunk 4]
```diff
@@ -63,7 +63,7 @@
                 </div>
 
                 <button type="submit" class="button">Save and continue</button>
-                <a asp-route="@RouteNames.Dashboard_Index_Get" class="button-link">Cancel</a>
+                <a asp-route="@RouteNames.Dashboard_Index_Get" asp-route-statusFilter="@Context.Request.Cookies[CookieNames.VacancyStatusFilter]" class="button-link">Cancel</a>
```
Comment (chrisjensenuk) [inline src/Employer/Employer.Web/Views/Part1/Employer/Employer.cshtml]: Can we add this as a property on the ViewModel so we're not aware of context?

File: src/Employer/Employer.Web/Controllers/DashboardController.cs [hunk 5]
```diff
@@ -1,31 +1,46 @@
...
+        public async Task<IActionResult> Dashboard([FromRoute]string employerAccountId, [FromQuery]string statusFilter)
         {
             var vm = await _orchestrator.GetDashboardViewModelAsync(employerAccountId);
             
             if (TempData.ContainsKey(TempDataKeys.DashboardErrorMessage))
             {
                 vm.WarningMessage = TempData[TempDataKeys.DashboardErrorMessage].ToString();
             }
+            
+            if (!string.IsNullOrEmpty(statusFilter) && Enum.TryParse<VacancyStatus>(statusFilter, out var filter))
```
Comment (chrisjensenuk) [inline src/Employer/Employer.Web/Controllers/DashboardController.cs]: Maybe move this if/else into its own method / extension method

File: src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml (hunk 3 above)
Comment (sachpatel) [reply src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml]: It is used to display `Not available` for any vacancy that hasn't has a ref no assigned.

File: src/Employer/Employer.Web/Views/Part1/Employer/Employer.cshtml (hunk 4 above)
Comment (sachpatel) [reply src/Employer/Employer.Web/Views/Part1/Employer/Employer.cshtml]: Don't want the ViewModel to be aware of context.

File: src/Employer/Employer.Web/Controllers/DashboardController.cs (hunk 5 above)
Comment (sachpatel) [reply src/Employer/Employer.Web/Controllers/DashboardController.cs]: Have moved into a method.

File: src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml (hunk 3 above)
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml]: I meant I don't see much enough improvement to warrant the creation of a new variable:
`<span asp-condition="@isMissingVacancyReferenceNo">Not available</span>`
vs
`<span asp-condition="@!vacancy.HasVacancyReference">Not available</span>`

File: src/Employer/Employer.Web/Views/Part1/Employer/Employer.cshtml (hunk 4 above)
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Views/Part1/Employer/Employer.cshtml]: It won't be. The controller will be and just sets the enum on the VM

File: src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml (hunk 3 above)
Comment (sachpatel) [reply src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml]: I am not going to change this. I think it reads cleaner and this is subjective.

Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Views/Dashboard/Dashboard.cshtml]: 👍

**Pull request merged**